app/views/projects/samples/attachments/_table.html.erb

```diff
@@ -1,113 +1,122 @@
+        <% end %>
+        <div class="flex items-center space-x-2 ml-auto">
+          <% if Flipper.enabled?(:sample_attachments_searching) %>
+            <%= render SearchComponent.new(
+              query: @q,
+              search_attribute: :puid_or_file_blob_filename_cont,
+              placeholder: t(:".search.placeholder"),
+              url: namespace_project_sample_path(format: :turbo_stream),
+              data: {
+                controller: "selection",
+                selection_storage_key_value: "files-#{@sample.id}"
+              }
+              ) do %>
+              <%= hidden_field_tag :limit, @pagy.limit %>
+              <%= hidden_field_tag :tab, @tab %>
             <% end %>
           <% end %>
-        </tbody>
-      </table>
-    </div>
-  <% else %>
-    <% if @allowed_to[:update_sample] %>
-      <%= render(
-        Viral::EmptyStateComponent.new(
-          icon_name: :document_text,
-          title: t(".empty_state.title"),
-          description: t(".empty_state.description"),
-          action_text: t(".empty_state.action_text"),
-          action_path:
-            new_namespace_project_sample_attachment_path(sample_id: @sample.id),
-          action_method: :get,
+          <%= link_to t("projects.samples.show.new_attachment_button"),
+          new_namespace_project_sample_attachment_path(sample_id: @sample.id),
           data: {
             turbo_frame: "sample_modal",
             turbo_stream: true,
           },
-        ),
-      ) %>
-    <% else %>
-      <%= render(
-        Viral::EmptyStateComponent.new(
-          icon_name: :document_text,
-          title: t(".empty_state.title"),
-          description: t(".empty_state.no_permission_description"),
-        ),
+          class: "button button--size-default button--state-default" %>
+          <%= button_to t("projects.samples.show.concatenate_button"),
+          new_namespace_project_sample_attachments_concatenation_path(
+            sample_id: @sample.id,
+          ),
+          method: :get,
+          data: {
+            turbo_frame: "sample_modal",
+            turbo_stream: false,
+            controller: "action-button",
+            action_button_required_value: 2,
+          },
+          class: "button button--size-default button--state-default action-button" %>
+          <%= button_to t("projects.samples.show.delete_files_button"),
+          new_namespace_project_sample_attachments_deletion_path(sample_id: @sample.id),
+          method: :get,
+          data: {
+            turbo_frame: "sample_modal",
+            turbo_stream: false,
+            controller: "action-button",
+            action_button_required_value: 1,
+          },
+          class: "button button--size-default button--state-default action-button" %>
+        </div>
+      </div>
+    <% end %>
+    <%= turbo_frame_tag "selected" %>
+    <%= turbo_frame_tag "sample_attachments" do %>
+      <%= render Attachments::TableComponent.new(
```

Comment:
The `render Attachments::TableComponent` is the same as in `app/views/projects/samples/attachments/index.turbo_stream.erb`. Can you move this code into a shared file to be used in multiple locations